cookbooks/aws-parallelcluster-common/spec/unit/resources/lustre_setup_spec.rb

```diff
@@ -0,0 +1,246 @@
+      end
+      Lustre.setup(runner)
+    end
+
+    it 'can not install lustre' do
+      is_expected.to write_log("FSx for Lustre is not supported in this RHEL version 8.1, supported versions are >= 8.2")
+        .with(level: :warn)
+    end
+  end
+
+  context "on redhat with kernel 4.18.0-425.3.1.el8" do
+    let(:chef_run) do
+      runner = ChefSpec::Runner.new(
+        platform: 'redhat', version: '8',
+        step_into: ['lustre']
+      ) do |node|
+        node.override['cluster']['platform_version'] = "8.2"
+        node.override['cluster']['kernel_release'] = "anything 4.18.0-425.3.1.el8 something"
+      end
+      Lustre.setup(runner)
+    end
+
+    it 'can not install lustre' do
+      is_expected.to write_log("FSx for Lustre is not supported in kernel version 4.18.0-425.3.1.el8 of RHEL, please update the kernel version")
+        .with(level: :warn)
+    end
+  end
+
+  context "on redhat supporting lustre" do
+    let(:chef_run) do
+      runner = ChefSpec::Runner.new(
+        platform: 'redhat', version: '8',
+        step_into: ['lustre']
+      ) do |node|
+        node.override['cluster']['platform_version'] = "8.2"
+        node.override['cluster']['kernel_release'] = "supported"
+      end
+      Lustre.setup(runner)
+    end
+
+    it 'installs lustre packages from repository and installs kernel module lnet' do
+      is_expected.to create_yum_repository("aws-fsx")
+        .with(baseurl: 'https://fsx-lustre-client-repo.s3.amazonaws.com/el/8/$basearch')
+        .with(gpgkey: 'https://fsx-lustre-client-repo-public-keys.s3.amazonaws.com/fsx-rpm-public-key.asc')
+        .with(retries: 3)
+        .with(retry_delay: 5)
+
+      is_expected.to run_execute('yum-config-manager_skip_if_unavail')
+        .with(command: "yum-config-manager --setopt=\*.skip_if_unavailable=1 --save")
+
+      is_expected.to install_package(%w(kmod-lustre-client lustre-client dracut))
+        .with(retries: 3)
+        .with(retry_delay: 5)
+
+      is_expected.to install_kernel_module("lnet")
+    end
+  end
+
+  for_oses([
+     %w(ubuntu 18.04),
+     %w(ubuntu 20.04),
+  ]) do |platform, version|
+    context "on #{platform}#{version}" do
+      let(:chef_run) do
+        runner = ChefSpec::Runner.new(
+          platform: platform, version: version,
+          step_into: ['lustre']
+        ) do |node|
+          node.override['cluster']['kernel_release'] = 'kernel_release'
+        end
+        Lustre.setup(runner)
+      end
+
+      it 'installs lustre packages from repository and installs kernel module lnet' do
+        is_expected.to add_apt_repository('fsxlustreclientrepo')
+          .with(uri: 'https://fsx-lustre-client-repo.s3.amazonaws.com/ubuntu')
+          .with(components: %w(main))
+          .with(key: ["https://fsx-lustre-client-repo-public-keys.s3.amazonaws.com/fsx-ubuntu-public-key.asc"])
+          .with(retries: 3)
+          .with(retry_delay: 5)
+
+        is_expected.to periodic_apt_update('')
```

Comment:
praise: cool! I didn't knew `periodic` was the default, wondering if we should explicit it somewhere